src/caselawclient/models/utilities/move.py

```diff
@@ -0,0 +1,142 @@
+import xml.etree.ElementTree as ET
+from typing import Any, Optional
+
+import ds_caselaw_utils as caselawutils
+
+from caselawclient.errors import MarklogicAPIError
+from caselawclient.models.judgments import Judgment
+from caselawclient.models.utilities.aws import copy_assets
+
+
+class NeutralCitationToUriError(Exception):
+    pass
+
+
+class OverwriteJudgmentError(Exception):
+    pass
+
+
+class MoveJudgmentError(Exception):
+    pass
+
+
+def overwrite_document(
+    source_uri: str,
+    target_citation: str,
+    api_client: Any,
+) -> str:
+    """Move the judgment at old_uri on top of the new citation, which must already exist
+    Compare to update_document_uri
+
+    :param source_uri: The URI with the contents of the document to be written. (possibly a failure url)
+    :param target_citation: The NCN (implying a URL) whose contents will be overwritten
+    :param api_client: An instance of MarklogicApiClient used to make the search request
+    :return: The URL associated with the `target_citation`
+    """
+
+    new_uri: Optional[str] = caselawutils.neutral_url(target_citation.strip())
+
+    if new_uri == source_uri:
+        raise RuntimeError(f"trying to overwrite yourself, {source_uri}")
+    if new_uri is None:
+        raise NeutralCitationToUriError(
+            f"Unable to form new URI for {source_uri} from neutral citation: {target_citation}"
+        )
+    if not api_client.document_exists(new_uri):
+        raise OverwriteJudgmentError(
+            f"The URI {new_uri} generated from {target_citation} does not already exist, so cannot be overwritten"
+        )
+    old_judgment = Judgment(source_uri, api_client)
+    try:
+        old_judgment_bytes = old_judgment.content_as_xml
+        old_judgment_xml = ET.XML(bytes(old_judgment_bytes, encoding="utf-8"))
+        api_client.save_judgment_xml(
+            new_uri,
+            old_judgment_xml,
+            annotation=f"overwritten from {source_uri}",
+        )
+        set_metadata(source_uri, new_uri, api_client)
+        # TODO: consider deleting existing public assets at that location
+        copy_assets(source_uri, new_uri)
+        api_client.set_judgment_this_uri(new_uri)
+    except MarklogicAPIError as e:
+        raise OverwriteJudgmentError(
+            f"Failure when attempting to copy judgment from {source_uri} to {new_uri}: {e}"
+        )
+
+    try:
+        api_client.delete_judgment(source_uri)
+    except MarklogicAPIError as e:
+        raise OverwriteJudgmentError(
+            f"Failure when attempting to delete judgment from {source_uri}: {e}"
+        )
+
+    return new_uri
+
+
+def update_document_uri(source_uri: str, target_citation: str, api_client: Any) -> str:
```

Comment:
isn't `overwrite_document_uri` essentially a superset of what update_document_uri can do?

i.e `update_document_uri` is redundant?